### PR TITLE
Refactor/props to payload

### PR DESCRIPTION
**Pull request by paasovaara**
Intermediate result of a larger refactoring: creating a new micro-service for StopEstimates. https://github.com/HSLdevcom/transitdata-tripupdate-processor/issues/40

This removes internal data models and uses the defined protobuf-models from commons. Has one intermediate datamodel PubtransData which converts the data coming from PubtransSource to StopEstimates. Then once StopEstimate-service has been built this app it's easy to just read that model directly when all internal processing already uses that data-model.

Also pushed lots of test-code upstream to commons so we can reuse them in integration tests.

Minor internal refactoring: use pulsar message key (String) as cache key, not dvj-id explicitly. more flexible for future use with trains and metros.

requires release of commons 0.8.1

### CODE REVIEW

File: src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java [hunk 1]
```diff
@@ -18,63 +18,75 @@
...
+            case SKIPPED:
                 stopTimeUpdateBuilder.setScheduleRelationship(GtfsRealtime.TripUpdate.StopTimeUpdate.ScheduleRelationship.SKIPPED);
                 break;
-            case Scheduled:
+            case SCHEDULED:
                 stopTimeUpdateBuilder.setScheduleRelationship(GtfsRealtime.TripUpdate.StopTimeUpdate.ScheduleRelationship.SCHEDULED);
                 break;
         }
+        // GTFS-RT treats times in seconds
+        long stopEventTimeInSeconds = stopEstimate.getEstimatedTimeUtcMs() / 1000;
```
Comment (kenkus-futurice) [inline src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java]: Scary, need to keep this conversion in mind.

File: src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java [hunk 2]
```diff
@@ -18,63 +18,75 @@
...
+        final int direction = PubtransFactory.joreDirectionToGtfsDirection(estimate.getTripInfo().getDirectionId());
 
         GtfsRealtime.TripDescriptor tripDescriptor = GtfsRealtime.TripDescriptor.newBuilder()
-                .setRouteId(reformatRouteName(event.getRouteData().getRouteName()))
-                .setDirectionId(event.getRouteData().getDirection())
-                .setStartDate(event.getRouteData().getOperatingDay())
-                .setStartTime(event.getRouteData().getStartTime())
+                .setRouteId(routeName)
+                .setDirectionId(direction)
+                .setStartDate(estimate.getTripInfo().getOperatingDay()) // Local date as String
```
Comment (kenkus-futurice) [inline src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java]: I wonder what the date is here after midnight?

File: src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    public InternalMessages.StopEstimate toStopEstimate() throws Exception {
+        InternalMessages.StopEstimate.Builder builder = InternalMessages.StopEstimate.newBuilder();
+        builder.setSchemaVersion(builder.getSchemaVersion());
+
+        InternalMessages.TripInfo.Builder tripBuilder = InternalMessages.TripInfo.newBuilder();
+        tripBuilder.setTripId(Long.toString(tripInfo.getDvjId()));
+        tripBuilder.setOperatingDay(tripInfo.getOperatingDay());
+        tripBuilder.setRouteId(tripInfo.getRouteId());
+        tripBuilder.setDirectionId(tripInfo.getDirectionId());//Jore format
```
Comment (kenkus-futurice) [inline src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java]: Does it make sense to call `direction_id` `jore_direction_id` in the `internal-messages.proto`?

File: src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        tripBuilder.setStartTime(tripInfo.getStartTime());
+        builder.setTripInfo(tripBuilder.build());
+        builder.setStopId(tripInfo.getStopId());
+        builder.setStopSequence(common.getJourneyPatternSequenceNumber());
+
+        InternalMessages.StopEstimate.Status scheduledStatus = (common.getState() == 3L) ?
```
Comment (kenkus-futurice) [inline src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java]: Is there any good way of making the `common.getState()` (and other proto enums) actual enums?

File: src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java (hunk 2 above)
Comment (paasovaara) [reply src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java]: at least according to [this](https://developers.google.com/transit/gtfs/reference/#frequenciestxt) also gtfs supports times > 24:00. I assume the operating day is then the previous date. These come directly from pubtrans in our case

File: src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java (hunk 1 above)
Comment (paasovaara) [reply src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java]: ours always in ms, public in secs

File: src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java (hunk 3 above)
Comment (paasovaara) [reply src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java]: this was discussed but there wasn't a clear call which was better... jore is quite context-specific term also.. decided that we leave it to comments and convention. time will tell if that was a good decision.

File: src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java (hunk 4 above)
Comment (paasovaara) [reply src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java]: Proto enums are found as java enums also, but this particular case (common.getState()) is a direct value from a database. We could of course convert that to string/enum in the SQL but the reason for 3L was the original: we wanted to keep the raw-data as-is. This demand however is slipping away so we could convert the code to more readable format. 

also the parsing goes away to StopEstimates-microservice so TripUpdate actually can work on better abstractions.

**Pull request merged**